### ISSUE TITLE
An unhandled exception of type 'System.NotSupportedException' occurred in System.Numerics.Vectors.dll

### ISSUE THREAD

**Issue by My-Khan** [zhongkaifu/RNNSharp#14]
Hello, First i really appreciate your work. My programming skills in C# is not good therefore apology in advance if question seems to your good-self stupid.
when i copile the source i got the error in ModelSetting .cs
System.NotSupportedException was unhandled
  HResult=-2146233067
  Message=Vector<T>.Count cannot be called via reflection when intrinsics are enabled.
  Source=System.Numerics.Vectors
  StackTrace:
       at System.Numerics.Vector`1.get_Count()
       at RNNSharp.ModelSetting.DumpSetting() in D:\RNNSharp-master\RNNSharp\ModelSetting.cs:line 51
       at RNNSharpConsole.Program.Train() in D:\RNNSharp-master\RNNSharpConsole\Program.cs:line 444
       at RNNSharpConsole.Program.Main(String[] args) in D:\RNNSharp-master\RNNSharpConsole\Program.cs:line 277
       at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
       at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
       at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ThreadHelper.ThreadStart()
  InnerException: 
here is my screen shoot:
![image](https://cloud.githubusercontent.com/assets/17508897/14415210/08509600-ffb6-11e5-9c8a-fe38f9b140cc.png)

so please help , i am new to C# and my work is pending since two days due to this bug. thanks

**Comment by bratao** [comment]
Are you compiling it as "Debug" ?
It seems to be a known bug in Debug mode (https://social.msdn.microsoft.com/Forums/vstudio/en-US/fd0a9ecc-b122-4795-a0e0-1cb6dc7a1a4a/anyone-got-simd-working-in-vs2015-net-46?forum=csharpgeneral)

**Comment by My-Khan** [comment]
Yes brataoi am compiling it in "Debug" mode

**Comment by My-Khan** [comment]
@bratao i followed your mentioned link, where i found that it is still unsolved issue. am i right?

**Comment by zhongkaifu** [comment]
@My-Khan  Did you try to run it "Release" mode ?
RNNSharp uses this Vector<T> lib to speed up encoding and decoding by SIMD instruction. As far as I know, if your CPU doesn't support AVX2 instruction, it will throw out an exception in "debug" mode.

I'm thinking about adding a compiling definition, such as _SUPPORT_SIMD_, for the machine that doesn't support AVX2, but currently, I'm focusing on deep recurrent neural netowork for RNNSharp, so this fixing will be implemented later.

**Comment by My-Khan** [comment]
many thanks sir for your kind reply. Now as per your kind suggestions i run it in release mode and got the following error message.
An unhandled exception of type 'System.Exception' occurred in RNNSharp.dll

Additional information: The offset of run time feature should be negative.
screen shots of my error are below.

![image](https://cloud.githubusercontent.com/assets/17508897/14451196/885674ee-009e-11e6-836a-d011db564edb.png)

![image](https://cloud.githubusercontent.com/assets/17508897/14451224/bc293676-009e-11e6-9e3b-3b7ddba44f7a.png)

**Comment by zhongkaifu** [comment]
Could you please share your configuration file ? It seems incorrect settings in real time feature.

**Comment by My-Khan** [comment]
Sir, below is my configuration file contents
#Template Features
TFEATURE_FILENAME:tfeature
#Context Features
TFEATURE_CONTEXT:-2,-1,0,1,2
#The context range for word embedding. In below, the context is current token, previous token and next token
#The word embedding data file name generated by WSDSharp
WORDEMBEDDING_FILENAME:myvector
WORDEMBEDDING_CONTEXT: -1,0,1
#The column index applied word embedding feature
WORDEMBEDDING_COLUMN: 0
#The run time feature
RTFEATURE_CONTEXT: 0

**Comment by zhongkaifu** [comment]
RTFEATURE_CONTEXT value should be negative( < 0), since RNNSharp uses previous tokens' outputs as run time features for current token.

In addition, this feature is only for forward neural network so far, bi-directional RNN doesn't support it.

**Comment by My-Khan** [comment]
Well sir the previous issue resolved, but a new issue generated, please also look this one, thanks in advance. this time it does not recognize the mapping tags. 
My training data is organized as, first column represent token,2nd column represents part of speech tag and last column represents label.
Below is the error message:
An unhandled exception of type 'System.DataMisalignedException' occurred in RNNSharp.dll

Additional information: Error: tag B_LOCATION is unknown. Tokens list: <s>  <s> SentBE  

contents of my mapfile are 
00  O
01  S_PERSON
02  B_PERSON
03  M_PERSON
04  E_PERSON
05  S_LOCATION
06  B_LOCATION
07  M_LOCATION
08  E_LOCATION
09  S_ORGANIZATION
10  B_ORGANIZATION
11  M_ORGANIZATION
12  E_ORGANIZATION
13  S_DESIGNATION
14  B_DESIGNATION
15  M_DESIGNATION
16  E_DESIGNATION

**Comment by zhongkaifu** [comment]
The tag file format has been updated. Each line is one tag, so you could remove id from the file and then try it again.

**Comment by My-Khan** [comment]
Well done sir, now it works.... one last query: i am still confused about validation file data.
what will be the validation file data?
1. are it is the same data  that is already exists in training file? OR
2. are it is the same data that will be used in testing phase?
3. are it is the totally separate portion of data that is neither used in training nor in testing, if Yes that what will be the proportion?
Thanks , looking for your kind reply

**Comment by zhongkaifu** [comment]
Validation set is usually used to choose best hyper-parameter. It should not be mixed with either training corpus or test corpus. For entire training set, you could randomly sample 20% data as validation set in each iteration.

One suggestion: For other questions unrelated to title, it would be better to open a new thread about it.

**Comment by My-Khan** [comment]
Many thanks for very clear answer.
i will try of my level best to follow your given suggestion in future.
Looking forward for your next  deep recurrent neural netowork for RNNSharp.
